harness/determined/experimental/core_v2/_core_v2.py

```diff
@@ -279,3 +296,44 @@ def url_reverse_webui_exp_view() -> str:
 
     assert _client is not None
     return core_v2._url_reverse_webui_exp_view(_client, exp_id)
+
+
+def _merge_config(
+    config: Optional[Config],
+    defaults: Optional[DefaultConfig],
+    unmanaged: Optional[UnmanagedConfig],
+) -> Config:
+    if defaults is not None or unmanaged is not None:
+        _show_deprecated_msg()
+    if config is not None:
+        return config
+    info = determined.get_cluster_info()
+    if defaults is None and info is None:
+        raise NotImplementedError(
```

Comment:
nit: i know this was here before, but i don't think `NotImplementedError` is technically/pythonically the right exception to throw here. prolly [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) is a better fit. not important though.